crates/rust/src/lib.rs

```diff
@@ -1248,6 +1248,15 @@ fn wasm_type(ty: WasmType) -> &'static str {
         WasmType::I64 => "i64",
         WasmType::F32 => "f32",
         WasmType::F64 => "f64",
+        WasmType::Pointer => "*mut ::core::ffi::c_void",
+        WasmType::Length => "usize",
+
+        // `PointerOrI64` can hold either a `u64` or a pointer with provenance.
+        // Neither a `u64` nor a pointer type can portably do both, so we use
+        // `MaybeUninit<u64>`, since `MaybeUninit` is [documented] to preserve
+        // provenance.
+        // [documented]: https://github.com/rust-lang/rfcs/blob/master/text/3559-rust-has-provenance.md#reference-level-explanation
```

Comment:
My read of the text in the RFC here is that `MaybeUninit<T>` has the same provenance of `T` where `u64` has no provenance here, but do you know if the RFC text meant differently?

Either way though I don't know of anything better so I think it's also ok to leave as-is.

Reply:
Ah ok then in that case it sounds like exactly what we want here 👍